docs/roadmap.md

```diff
@@ -4,6 +4,149 @@ title: Babel Roadmap
+
+### Move `@babel/preset-env` into `@babel/core`
+
+A minimal Babel transforming setup requires at least three packages:
+- `@babel/core`
+- `@babel/preset-env`
+- a Babel "runner" (`@babel/cli`, `babel-loader`, `@rollup/plugin-babel`, etc)
+
+Moving `@babel/preset-env` directly into `@babel/core` has two big advantages:
+- It will be easier to configure Babel in simple projects, you would only need to enable a `compileJS: true` option in `babel.config.json` (or it could even be the default in the future -- it can't be default as `@babel/eslint-parser` does not compile the source)
+- It will make sure that the plugin versions are in sync with the `@babel/core` version, avoiding most of the bugs casued by accidentally incompatible packages versions
+- When we'll start moving to ESM, it will be hard to resolve and load plugin synchronously in `transformSync`. This prevents it from being a problem.
+
+There is already [a RFC](https://github.com/babel/rfcs/pull/10) to move _plugins_ for stable ECMAScript features in `@babel/core`, which is the first step in this direction.
+
+With our current `@babel/preset-env` architecture, we would need to specially handle official plugins to automatically enable or disable them based on `targets`.
+However, this has two drawbacks:
+- Compatibility data for a specific plugin is completely separated from the plugin implementation (it's not even a dependency, more something like an internal implicit peer dependency: plugin -> @babel/core -> @babel/compat-data);
+- Official plugins would get a special treatment from `@babel/core`, while we want to make sure that third-party plugins have the same capabilities as official plugins.
+
+### Continue developing the `babel-polyfills` project
+
+We have already decided to remove the old `core-js@2` support from `@babel/preset-env` in Babel 8, and we also want to stop promoting a specific third-party polyfill giving our users the impression that it's part of Babel itself.
+
+This might happen in two different ways:
+- We just remove `core-js@3` from `@babel/preset-env` in Babel 8, encouraging users to migrate to `babel-plugin-polyfill-corejs3` (which is what `@babel/preset-env` internally uses starting from version 7.10.0)
+- We can keep `core-js@3` support in `@babel/preset-env`, but not migrate it to `@babel/core` when we'll move the transform plugins.
+
+Whatever path we take, we would like to offer at least one alternative to our users when they'll need to update the `core-js` integration in their configuration. `core-js` is a really good polyfill that ensures the highest possible spec compliancy, but different users might prefer different trade-offs.
+
+([Nicolò](https://github.com/nicolo-ribaudo)) is working with [@ljharb](https://github.com/ljharb) to make sure that the [`@es-shims` project](https://github.com/es-shims/) supports at least all the ES2015+ features (we actually aim for ES5+), so that Babel users are free to choose to between at least two alternatives.
```

Comment:
```suggestion
([Nicolò](https://github.com/nicolo-ribaudo)) is working with [@ljharb](https://github.com/ljharb) to make sure that the [`@es-shims` project](https://github.com/es-shims/) supports at least all the ES2015+ features (we actually aim for ES5+), so that Babel users are free to choose to between at least two options.
```